js/drafts/composer.js

```diff
@@ -155,8 +155,15 @@ function Composer(newRecords, account, identity) {
 
   body.attachments.forEach(function(attachment) {
     try {
-      var attachmentNode = new MimeNode(attachment.type,
-                                        { filename: attachment.name });
+      var attachmentNode = new MimeNode(
+        attachment.type,
+        {
+          // This implies Content-Disposition: attachment
+          filename: attachment.name
+        });
+      // Explicitly specify base64 encoding; this is the default for non
+      // text/* parts, but it's important for us to not break on those, too.
+      attachmentNode.setHeader('Content-Transfer-Encoding', 'base64');
```

Comment:
Uh, and my comment here is insufficiently correct.  Our attachment logic precomputes base64-encoded blobs.  So the attachment content is base64 encoded, so we need to say this anyways.  But it's also the root of the breakage where we just show our replacement marker.  The actual comment I'm revising now will be sufficiently correct :)